tests/test_csv_readers.py

```diff
@@ -0,0 +1,857 @@
+                f"Expected {coord_name} to have values {expected_values} in order, "
+                f"got {list(data.coords[coord_name].values)}"
+            )
+        else:
+            assert set(data.coords[coord_name].values) == set(expected_values), (
+                f"Expected {coord_name} to have values {expected_values}, "
+                f"got {list(data.coords[coord_name].values)}"
+            )
+
+
+def assert_single_coordinate(data, selection, expected):
+    """Assert values for a single coordinate selection match expected values."""
+    actual = data.sel(**selection).data_vars
+    for k, v in expected.items():
+        if isinstance(v, float):
+            assert np.isclose(actual[k].item(), v), (
+                f"Expected {k} to be {v}, got {actual[k].item()}"
+            )
+        else:
+            assert actual[k].item() == v, (
+                f"Expected {k} to be {v}, got {actual[k].item()}"
+            )
+
+
+@fixture
+def timeslice():
+    """Sets up global timeslicing scheme to match the default model."""
+    from muse.timeslices import setup_module
+
+    timeslice = """
+    [timeslices]
+    all-year.all-week.night = 1460
+    all-year.all-week.morning = 1460
+    all-year.all-week.afternoon = 1460
+    all-year.all-week.early-peak = 1460
+    all-year.all-week.late-peak = 1460
+    all-year.all-week.evening = 1460
+    """
+
+    setup_module(timeslice)
+
+
+@fixture
+def model_path(tmp_path, timeslice):
+    """Creates temporary folder containing the default model."""
+    examples.copy_model(name="default", path=tmp_path)
+    return tmp_path / "model"
+
+
+@fixture
+def timeslice_model_path(tmp_path, timeslice):
+    """Creates temporary folder containing the default model."""
+    examples.copy_model(name="default_timeslice", path=tmp_path)
+    return tmp_path / "model"
+
+
+@fixture
+def trade_model_path(tmp_path):
+    """Creates temporary folder containing the trade model."""
+    examples.copy_model(name="trade", path=tmp_path)
+    return tmp_path / "model"
+
+
+@fixture
+def correlation_model_path(tmp_path):
+    """Creates temporary folder containing the correlation model."""
+    examples.copy_model(name="default_correlation", path=tmp_path)
+    return tmp_path / "model"
+
+
+def test_read_global_commodities(model_path):
+    from muse.readers.csv import read_global_commodities
+
+    path = model_path / "GlobalCommodities.csv"
+    data = read_global_commodities(path)
+
+    # Check properties of the dataset
+    assert_schema(
+        data,
+        DatasetSchema(
+            dims={"commodity"},
+            coords={"commodity": CoordinateSchema(dims=("commodity",), dtype="object")},
+            data_vars={
+                "comm_name": "object",
+                "comm_type": "object",
+                "emmission_factor": "float64",
+                "heat_rate": "int64",
+                "unit": "object",
+            },
+        ),
+    )
```

Comment:
After seeing that you have `generate_dataarray_schema` below, and if you consider my suggestion to make them classmethods, I wonder if it wont be simpler to replace this `assert` schema by:

```python
assert DatasetSchema.from_ds(data) == DatasetSchema(...)
```

It does not save you much - if anything - but as dataclasses have built-in equality comparison, we can very well use them. 

Also, from the point of view of readability, I feel that defining the schema and then doing the comparison would be better:

```python
global_commmodities_schema = DatasetSchema(...)
assert DatasetSchema.from_ds(data) == global_commmodities_schema
```